### ISSUE TITLE
Recreating the Configuration, any suggestions?

### ISSUE THREAD

**Issue by CallumCarmicheal** [juryben/XboxKeyboardMouse#97]
Im in the process of recreating the configuration system while working on the new GUI. As of the current system we are using INI's which are not serialized and this makes it a pain to use while programming. 

Im thinking of recreating the config to make it look and seem cleaner, are there any suggestions in which format we should use. Im currently thinking json or yml.

**Comment by CallumCarmicheal** [comment]
Im currently leaning towards yaml, any one got any objections to this?

**Comment by fqlx** [comment]
There's generally more support for JSON so I'm a JSON fan.

**Comment by CallumCarmicheal** [comment]
@fqlx i know both was just curious on input from the community but it seems they dont mind. I guess we can go with json.

**Comment by DavidRieman** [comment]
I know .NET does good XML serialization out of the box which can be utilized with minimal effort.  I was able to do it quickly per discussion at https://github.com/fqlx/XboxKeyboardMouse/issues/60 but backed out.  If we're seriously considering the ClickOnce approach (with a likely-required update to _where_ things get saved anyway), IMO that would be a great opportunity to switch formats too.  :)

**Comment by CallumCarmicheal** [comment]
@DavidRieman Im not quite thinking about how hard it is to implement because almost all configuration libraries support serialization its more user friendly should they need to manually edit the config for what ever reason, does ClickOnce support nuget?

**Comment by DavidRieman** [comment]
Yes, ClickOnce handles dependencies that were fetched through NuGet.
